crates/core/component/chain/src/component/app_hash.rs

```diff
@@ -106,36 +110,43 @@ impl AppHashRead for Snapshot {
         };
 
         Ok((
-            value,
+            some_value,
             MerkleProof {
-                proofs: vec![jmt_commitment_proof, root_proof],
+                proofs: vec![membership_proof, root_proof],
             },
         ))
     }
 
+    /// Returns the value and a proof of inclusion up to the current app hash,
+    /// using Tendermint's proof format. Returns `None` if the key is not present.
+    ///
+    /// Review: it looks like the Tendermint proof format does not support
```

Comment:
Reviewer scrutiny appreciated here, I'm not sure this is true (re: tendermint merkle proof format) - or that we want to reply to ABCI queries with nonexistence proofs regardless. @avahowell